uat/custom-components/src/main/java/com/aws/greengrass/artifacts/LogGenerator.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.artifacts;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.rolling.RollingFileAppender;
+import ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy;
+import ch.qos.logback.core.util.FileSize;
+import software.amazon.awssdk.aws.greengrass.model.InvalidArgumentsError;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/*
+This component rotates files with different name. Namely, the file is rotated and renamed by adding the timestamp and
+ some seq number to differ from others. The rolling policy is LogBack.TimeBasedRollingPolicy,
+ more information at https://logback.qos.ch/manual/appenders.html
+ */
+public class LogGenerator implements Consumer<String[]> {
+    private static final String rotationNamePattern = "_%d{yyyy-MM-dd_HH-mm}_%i";
+    private String logFileName;
+    private int writeFreqMs;
+    private int fileSizeBytes;
+    private int numberOfLogs;
+    private String logDirectory;
+
+
+    @Override
+    public void accept(String[] args) {
+        init(args);
+
+        try {
+            generateLogs();
+        } catch (Exception e) {
+            System.exit(0);
+        }
+    }
+
+    private void init(String[] args) {
+        logFileName = args[0];
+        fileSizeBytes = getFileSizeInBytes(Double.parseDouble(args[1]), args[2]);
+        writeFreqMs = Integer.parseInt(args[3]);
+        numberOfLogs = Integer.parseInt(args[4]);
+        logDirectory = args[5];
+
+        if (logDirectory.isEmpty()) {
+            throw new InvalidArgumentsError("LogDirectory is required");
+        }
+    }
+
+    private void generateLogs() throws InterruptedException {
+        Logger logger = configureLogger(logDirectory, logFileName, fileSizeBytes);
+
+        for (int i = 1; i <= numberOfLogs; i++) {
+            String logLine = String.format("(seq: %s, name: %s)", i, logFileName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: i (declared at line 61). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 62. Consider formatting this data with &#39;%d&#39; instead.